miden-lib/src/notes/mod.rs

```diff
@@ -88,3 +121,26 @@ pub fn notes_try_from_elements(elements: &[Word]) -> Result<NoteStub, NoteError>
 
     Ok(stub)
 }
+
+/// Utility function generating RECIPIENT for the P2ID note script created by the ASWAP script
+pub fn build_p2id_recipient(target: AccountId, serial_num: Word) -> Result<Digest, NoteError> {
+    let assembler = assembler();
+
+    let p2id_bytes = include_bytes!(concat!(env!("OUT_DIR"), "/assets/P2ID.masb"));
+
+    let note_script_ast =
+        ProgramAst::from_bytes(p2id_bytes).map_err(NoteError::NoteDeserializationError)?;
+
+    let (note_script, _) = NoteScript::new(note_script_ast, &assembler)?;
+
+    let script_hash = note_script.hash();
```

Comment:
It would be cool if we could do this only once rather than every time we build a note. Maybe there is a way to do this at build time? or maybe using lazy static initialization?

Reply:
Also is it okay to leave it here or should we define a new file to contain utility function for scripts?

Reply:
I don't see for now a way to do this at build time. We could use lazy static initialisation to initialise it only once. 

That would make the code more complex and less clean. I am not sure of the overhead of doing it multiple times.

Reply:
We don't have to do it in this PR, but let's create an issue to implement the lazy static approach. Script compilation (depending on complexity), could be non-negligible.